src/lightning/fabric/strategies/launchers/xla.py

```diff
@@ -100,12 +100,9 @@ def _wrapping_function(
     ) -> None:
         import torch_xla.core.xla_model as xm
 
+        # `get_xla_supported_devices` in the spawned process returns the logical devices (2 for v2/v3 and 1 for v4)
         if _using_pjrt() and len(xm.get_xla_supported_devices()) > 1:
-            # `get_xla_supported_devices` in the spawned process returns the logical devices (2 for v2/v3 and 1 for v4)
-            # so when there's more than one (multithreading), objects need to be deep-copied
-            import copy
-
-            function, args, kwargs = copy.deepcopy((function, args, kwargs))
```

Comment:
@Liyang90 made the deepcopy in a single call to "maintain the reference links between these objects" (https://github.com/Lightning-AI/lightning/pull/17352#discussion_r1166274656). This PR changes this as `apply_to_collection` will not do this. I'll add the TPU label to the title so TPU tests run.

Reply:
XLA uses threading here, so the purpose is different. I reverted the change. For the fix I'm doing here, deep-copying is not sufficient.